openedx/features/wikimedia_features/meta_translations/utils.py

```diff
@@ -0,0 +1,263 @@
+
+
+def set_course_translation(course_key, source_key):
+    """
+    updete course translation table
+    """
+    course_id = str(course_key)
+    base_course_id = str(source_key)
+    CourseTranslation.objects.create(course_id = course_id, base_course_id = base_course_id)
+
+
+def get_html_components_data(block):
+    """
+    Extract data from html type xblock componenets
+
+    Arguments:
+        block: html type course-outline block
+
+    Returns:
+        dict of extracted data
+    """
+    return {
+        "display_name": block.display_name,
+        "content": block.data
+    }
+
+
+def get_video_components_data(block):
+    """
+    Extract data from video type xblock components
+
+    Arguments:
+        block: video type course-outline block
+
+    Returns:
+        dict of extracted data
+    """
+    from xmodule.video_module.transcripts_utils import get_video_transcript_content
+    data = get_video_transcript_content(block.edx_video_id, block.transcript_language)
+    json_content = json.loads(data['content'].decode("utf-8"))
+
+    return {
+        "display_name": block.display_name,
+        "transcript": json_content.get('text')
+    }
+
+
+def get_component_info(component):
+    """
+    It will call relevant data retrieve method and return componenet level (problem, video, html etc) info retrieved
+    from given course-outline block.
+
+    Arguments:
+        component: componenet level course-outline block
+
+    Returns:
+        dict of extracted data
+    """
+    block = modulestore().get_item(component)
+    COMPONENTS_FUNCTION_MAPPING = {
+        'html': get_html_components_data,
+        'problem': get_html_components_data,
+        'video': get_video_components_data
+    }
+
+    return {
+        'id': str(component),
+        'type': component.block_type,
+        'data': COMPONENTS_FUNCTION_MAPPING[component.block_type](block)
+    }
+
+
+def get_module_info(module):
+    """
+    Returns module level (sections, subsections and units) info retrieved from given course-outline block.
+
+    Arguments:
+        module: module level course-outline block
+
+    Returns:
+        dict of extracted data
+    """
+    return {
+        'id': module['id'],
+        'type': module['category'],
+        'data': {
+            'display_name': module['display_name'],
+        }
+    }
+
+
+def map_base_course_block(existing_course_blocks, outline_block_dict, course_key):
+    """
+    Map base-course block -> Sync Course Outline block data to CourseBLock and CourseBLockData table
+    - if data does not exist, create new course-block and data entries.
+    - if data is updated in course outline, update data in relevant course-block.
+
+    Arguments:
+        existing_course_blocks (CourseBlock): existing course-blocks in db for given course-key
+        outline_block_dict (dict): contains data of course-outline block
+        course_key (CourseKey): course on which mapping needs to perform
+    """
+    try:
```

Comment:
can we narrow down the scope of this `try` block? wrap minimum statements in `try` block.